src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -146,40 +146,6 @@ public override IReadOnlyList<Location> AdditionalLocations
                 get { return _properties; }
             }
 
-            public override bool Equals(Diagnostic obj)
-            {
-                var other = obj as SimpleDiagnostic;
-                if (other == null)
-                {
-                    return false;
-                }
-
-                if (AnalyzerExecutor.IsAnalyzerExceptionDiagnostic(this))
```

Comment:
```DiagnosticDescriptor.GetHashCode()``` calls ```_description.GetHashCode()``` so different descriptions will potentially result in different hash codes, even though the containing Diagnostics compare Equals.